### DISCUSSION TITLE
Port 3000 problem

### DISCUSSION THREAD

**Discussion by indraromdoni** [devlikeapro/waha#71] [Q&A] [answered]
I can't run WAHA in another port instead port 3000. Can you tell me why? I need to run it in another port because the port 3000 I already use for another.
![IMG_20230302_151538](https://user-images.githubusercontent.com/72484049/222371962-90f49bed-1ab9-4804-9409-e14899522137.jpg)
![IMG_20230302_151603](https://user-images.githubusercontent.com/72484049/222371988-d00d7042-39f0-4c40-ae64-f70c520455ad.jpg)
![IMG_20230302_151618](https://user-images.githubusercontent.com/72484049/222372002-fc6a2b8b-c746-4e77-8571-5023a61a7787.jpg)
![IMG_20230302_151641](https://user-images.githubusercontent.com/72484049/222372017-06c63ffb-01e1-4828-92d6-ae3be3ee062b.jpg)

**Comment by allburov** [comment]
Try to set [environment variable WHATSAPP_API_PORT](https://waha.devlike.pro/docs/how-to/config/)
```bash
docker run -it -e "WHATSAPP_API_PORT=3020" devlikeapro/whatsapp-http-api
 
```

**Reply by indraromdoni** [reply]
I try your instruction but I still can not open with the browser.
![IMG_20230303_090824](https://user-images.githubusercontent.com/72484049/222614434-13f78b2d-d47a-4be7-a4f8-2e45babae819.jpg)
![IMG_20230303_090840](https://user-images.githubusercontent.com/72484049/222614442-0f2b827b-51ed-479d-8598-df84a53ae293.jpg)

**Comment by allburov** [comment]
If you just reused the command I gave - yep, it won't work, because you must map ports to localhost as well

So the whole command would be:
```bash
docker run -it -p 3000:3000/tcp -e "WHATSAPP_API_PORT=3020" devlikeapro/whatsapp-http-api
# OR 
docker run -it --network host-e "WHATSAPP_API_PORT=3020" devlikeapro/whatsapp-http-api
```

For the future type the commands that you run if you're asking questions, please.

**Comment by indraromdoni** [comment, marked as answer]
I found the solution. I trying this code below :

`docker run -it --rm -p 127.0.0.1:3020:3000/tcp --name whatsapp-http-api devlikeapro/whatsapp-http-api`

It will run in port 3020 instead of 3000.
Thank you for your help @allburov .
